libs/add_compat_headers.sh

```diff
@@ -0,0 +1,138 @@
+    if [[ script_sourced -eq 1 ]]; then
+        return
+    fi  
+fi
+
+function parse_arguments() {
+
+    # store arguments list
+    POSITIONAL=()
+
+    while [[ $# -gt 0 ]]
+    do
+        local key="$1"
+        case $key in
+            -m|--module)
+                module=$2
+                echo "module : ${module}"
+                shift # pass option
+                shift # padd value
+                ;;
+            -n|--new_path)
+                new_path=$2
+                echo "new_path : ${new_path}"
+                shift # pass option
+                shift # padd value
+                ;;
+            -o|--old_path)
+                old_path=$2
+                echo "old_path : ${old_path}"
+                shift # pass option
+                shift # padd value
+                ;;
+            -p|--project_path)
+                project_path=$2
+                echo "project_path: ${project_path}"
+                shift # pass option
+                shift # padd value
+                ;;
+            --help|*)
+                echo $"Usage: $0 [-m, --module <value>] [-o, --old_path <value>] \
+                [-n, --new_path <value>] [-p, --project_path <value>]"
+                echo "Example: "$0" -m cache -o hpx/util/cache -n hpx/cache"
+                echo "Can specify the --project_path if different from $HPX_ROOT"
+                return
+        esac
+    done
+
+    # restore positional parameters
+    set -- "${POSITIONAL[@]}"
+
+}
+
+# Default values which can be overwritten while parsing args
+project_path=$HPX_ROOT
+module=cache
+old_path=hpx/util/${module}
+new_path=hpx/${module}
+# Default is globbing compatibility files
+all_files=1
+files=
+
+# Parsing arguments
+parse_arguments "$@"
+
+# Usual vars (depend on the parsing step)
+libs_path=$project_path/libs
+module_path=$libs_path/${module}
+module_caps=${module^^}
+
+# Project path not set (full specified path to be sure which source is used)
+if [[ -z $HPX_ROOT && -z $project_path ]]; then
+    "HPX_ROOT env var doesn't exists and project_path option not specified !"
+    if [[ script_sourced -eq 1 ]]; then
+        return
+    fi  
+fi
+
+pushd $module_path/include_compatibility/${old_root}
+# To enable **
+shopt -s globstar
+# Globbing step to get all the include_compatibility files, the files have to
+# already be there, we are just rewriting them
+if [[ all_files -eq 0 ]]; then
+    # To set the files manually, don't forget to put the subdirs (of old_root) if there is
+    files=(histogram rolling_max rolling_min)
+    files=(${files[@]/%/.hpp})
+else
+    files=($(ls **/*.hpp))
+fi
+
+echo "Files overwritten :"
+# Create the corresponding compatibility headers
+# (no indentation in the for loop otherwise it appears in the cat)
+for full_file in "${files[@]}"; do
+# basename not used otherwise we lose hierarchy if any
+f=${full_file#"$old_path/"}
+echo $full_file
+cat >${full_file} <<EOL
+//  Copyright (c) 2019 Ste||ar Group
```

Comment:
Can you indent the commands above this line?

Reply:
yeah sure but it is a bit weird ^^